packages/browser/src/utils/survey-utils.ts

```diff
@@ -39,14 +41,61 @@ export const getSurveySeenKey = (survey: Pick<Survey, 'id' | 'current_iteration'
     return surveySeenKey
 }
 
-export const setSurveySeenOnLocalStorage = (survey: Pick<Survey, 'id' | 'current_iteration'>) => {
-    const isSurveySeen = localStorage.getItem(getSurveySeenKey(survey))
-    // if survey is already seen, no need to set it again
+/**
+ * We used to retrieve from localStorage directly. Now, we instead rely on the persistence API, since this is the preferred way to store data.
+ * But, since we might have customers that might be using surveys with persistence disabled, we need to fallback to localStorage
+ * to maintain backwards compatibility.
+ */
+export const getFromPersistenceWithLocalStorageFallback = (key: string, posthog?: PostHog) => {
+    try {
+        if (posthog?.persistence && !posthog.persistence.isDisabled()) {
+            return posthog.persistence.get_property(key)
+        }
+        return localStorage.getItem(key)
+    } catch (e) {
+        SURVEY_LOGGER.error('Error getting property from persistence or localStorage', e)
```

Comment:
those logs would be better if it was logged when an error happened within the persistence or localStorage happened so it'd be easier to debug issues.
right now we'd not know whats throwing, the persistence or localStorage

Reply:
separate try/catch blocks for all of them now